File: ios/Adyen2.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Adyen2, NSObject)

    RCT_EXTERN_METHOD(encryptCard:(NSDictionary *)params
                      resolver:(RCTPromiseResolveBlock)resolve
                      rejecter:(RCTPromiseRejectBlock)reject
                     )

@end

// #import "Adyen2.h"


// @implementation Adyen2

// RCT_EXPORT_MODULE()

// RCT_EXPORT_METHOD(sampleMethod:(NSString *)stringArgument numberParameter:(nonnull NSNumber *)numberArgument callback:(RCTResponseSenderBlock)callback)
// {
// 	callback(@[[NSString stringWithFormat: @"numberArgument: %@ stringArgument: %@", numberArgument, stringArgument]]);
// }

// @end

File: ios/Adyen2.swift
import Foundation
import Adyen

@objc(Adyen2)
class Adyen2: NSObject {
    @objc
    func encryptCard(
          _ cardInfo:NSDictionary,
          resolver resolve: RCTPromiseResolveBlock,
          rejecter reject: RCTPromiseRejectBlock
        ) -> Void {
            let card = CardEncryptor.Card(
                number: cardInfo["number"] as? String,
                securityCode: cardInfo["cvc"] as? String,
                expiryMonth: cardInfo["expiryMonth"] as? String,
                expiryYear: cardInfo["expiryYear"] as? String
            )
            
            let publicKey = cardInfo["publicKey"] as! String;
            
            let encryptedCard = CardEncryptor.encryptedCard(for: card, publicKey: publicKey)
            
            resolve([
                "number": encryptedCard.number,
                "expiryYear": encryptedCard.expiryYear,
                "expiryMonth": encryptedCard.expiryMonth,
                "cvc": encryptedCard.securityCode
            ])
    }
}
